nexus-repository-r/src/main/java/org/sonatype/nexus/repository/r/internal/RPathUtils.java

```diff
@@ -95,20 +95,17 @@ static String cutFilenameFromPath(final String path) {
 
   /**
    * Determines asset kind by it's path
-   *
-   * @param assetPath asset path
-   * @return R {@link AssetKind}
    */
-  static String getAssetKind(final String assetPath) {
-    String assetKind = ARCHIVE.name();
+  static AssetKind determineAssetKind(final String assetPath) {
     if (Pattern.compile(PATTERN_PACKAGES).matcher(assetPath).matches()) {
-      assetKind = PACKAGES.name();
+      return PACKAGES;
     }
     else if (Pattern.compile(PATTERN_METADATA_RDS).matcher(assetPath).matches()) {
-      assetKind = RDS_METADATA.name();
+      return RDS_METADATA;
+    }
+    else {
+      return ARCHIVE;
     }
```

Comment:
in NXRM mostly used the format like in the code before your changes. multiple returns increases complexity of a function

Reply:
done in 7a2327f